src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -744,7 +751,20 @@ public RepairRun addRepairRun(Builder repairRun, Collection<RepairSegment.Builde
 
   @Override
   public boolean updateRepairRun(RepairRun repairRun) {
-    session.execute(
+    return updateRepairRun(repairRun, Optional.of(true));
+  }
+
+  @Override
+  public boolean updateRepairRun(RepairRun repairRun, Optional<Boolean> updateRepairState) {
```

Comment:
curious: how is this not still implicitly a read-before-write anti-pattern?

i.e. how will the write know for certain that it is not updating state?

if it is because this method is only ever called by the "leader elect" then an `assert` statement would be nice here to ensure this really is the case of all scenarios…

Reply:
I made it so that we either update status in both tables or in none of them.

Reply:
Yes, and I see now the separate insert statement that doesn't touch the `state` column.

Very nice. It's only to look at the call hierarchy to `RepairRunner.updateLastEvent(..)` to see the positive impact this will have!